java/src/jmri/Section.java

```diff
@@ -1571,9 +1576,8 @@ private boolean setDirectionSensorByConnectivity(TrackNode tNode, TrackNode altN
                 tBlock = (tn == null) ? null : cUtil.getExitBlockForTrackNode(tn, null);
             }
             if (tBlock != null) {
-                lb = InstanceManager.getDefault(LayoutBlockManager.class).
-                        getByUserName(tBlock.getUserName());
-                if (lb != null) {
+                if (InstanceManager.getDefault(LayoutBlockManager.class).getByUserName(tBlock.getUserName()) != null) {
```

Comment:
That's a fine argument to keep this as is, and suppress the xBugs warning. Will review all.

Reply:
Instead of suppressing the warning, you could actually solve for it. Since tBlock.getUserName() can return different values in subsequent calls, it’s better to make one call to getUserName(), storing that in a variable, and if the variable is not null, use it. If we are just going to suppress legitimate SpotBugs warnings, we might as well not be using it. I would prefer to fix legitimate than suppress them.